### ISSUE TITLE
Uncaught exception when user with read but no write permissions tries to create transcript

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/Apollo#553]
I am posting this error for a specific case, but it is a general symptom of the following: We have many different ways that errors are propagated to the client, with practically each web service implementing the functionality itself, and that is if we are mindful enough to even propagate them at all. In the case of creating a transcript, a user with read but no write permissions will produce uncaught exceptions on the server side that are not seen on the client side, leaving them without a clue as to what happened.

Here is that specific case's server side error message

```
2015-08-26 17:26:56,259 [clientInboundChannel-4] ERROR springwebsocket.GrailsSimpAnnotationMethodMessageHandler  - Unhandled exception
java.lang.reflect.InvocationTargetException
    at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3_closure4_closure5.doCall(AnnotationEditorController.groovy:782)
    at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3_closure4.doCall(AnnotationEditorController.groovy:781)
    at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3.doCall(AnnotationEditorController.groovy:777)
    at groovyx.gpars.group.PGroup$3.call(PGroup.java:289)
    at groovyx.gpars.group.PGroup$4.run(PGroup.java:313)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
Caused by: org.bbop.apollo.AnnotationException: You have insufficent permissions [read < write] to perform this operation
    at org.bbop.apollo.PermissionService.$tt__checkPermissions(PermissionService.groovy:415)
    at org.bbop.apollo.RequestHandlingService.$tt__addTranscript(RequestHandlingService.groovy:632)
    ... 8 more
```

**Comment by cmdcolin** [comment]
I added some exception handling here

e6499338a601b

**Comment by nathandunn** [comment]
I agree .  . if we could link this with bootstrap alters somehow that would be perfect.

**Comment by nathandunn** [comment]
This is more of a generic UI / error-propagation issue without a specific fix.  Right now the behavior is that it does nothing, which is also acceptable.   We can fix these as we come across them as well.

**Comment by nathandunn** [comment]
I think we should look at this in 2.0.6.

**Comment by nathandunn** [comment]
Fixed as part of #1643
